src/Facebook/FileUpload/FacebookVideoUploader.php

```diff
@@ -0,0 +1,255 @@
+     */
+    public $endOffset;
+
+    /**
+     * @var int ID of the video
+     */
+    public $videoId;
+}
+
+/**
+ * Class FacebookVideoUploader.
+ */
+class FacebookVideoUploader
+{
+    /**
+     * @var VideoUploadContext
+     */
+    protected $context;
+
+    /**
+     * @var int Max retry times
+     */
+    protected $maxTransferTries;
+
+    /**
+     * @var Facebook Reference to Facebook
+     */
+    protected $facebook;
+
+    public function __construct(Facebook $fb)
+    {
+        $this->facebook = $fb;
+    }
+
+    public function setFacebook($fb)
+    {
+        $this->facebook = $fb;
+    }
+
+    /**
+     * Upload video file by chunks
+     *
+     * @param string       $filePath
+     * @param int          $targetId
+     * @param string       $accessToken
+     * @param int          $maxTransferTries
+     * @return array
+     */
+    public function upload($filePath, $targetId, $accessToken, $maxTransferTries = 3)
+    {
+        //validate params
+        $res = [];
+        if (empty($this->facebook) || empty($filePath) || empty($targetId) || empty($accessToken)) {
+            $res['errorMsg'] = 'Missing required params: ' .
+                (empty($this->facebook) ? 'facebook ' : '') .
+                (empty($accessToken) ? 'accessToken ' : '') .
+                (empty($filePath) ? 'filePath ' : '') .
+                (empty($targetId) ? 'targetId ' : '');
+
+            $res['success'] = false;
+            return $res;
+        }
+
+        //validate files
+        if (!file_exists($filePath)) {
+            $res['error'] = [
+                'message' => 'File not exists',
+            ];
+            $res['success'] = false;
+            return $res;
+        }
+
+        //set up context
+        $this->context = new VideoUploadContext();
+        $this->context->filePath = $filePath;
+        $this->context->fileSize = filesize($filePath);
+        $this->context->targetId = $targetId;
+        $this->context->accessToken = $accessToken;
+
+        //setup max retries
+        $this->maxTransferTries = $maxTransferTries;
+
+        try {
+            //Start - Initialize an upload session
+            $this->uploadStart();
+
+            //Transfer - Upload video chunks
+            $this->uploadTransfer();
+
+            //Finish - Post the video
+            $this->uploadFinish();
+            $res['success'] = true;
+            $res['video_id'] = $this->context->videoId;
+        } catch (FacebookResponseException $e) {
+            $res['success'] = false;
+            $res['error'] = isset($e->getResponseData()['error']) ? $e->getResponseData()['error'] : array();
+        } catch (FacebookSDKException $e) {
+            $res['error'] = [
+                'message' => $e->getMessage(),
+            ];
+            $res['success'] = false;
+        }
+
+        return $res;
+    }
+
+    /**
+     * Upload - phase start
+     */
+    public function uploadStart()
```

Comment:
What's the reason to mark this function as public? Maybe it is better to set it as private.